packages/@aws-cdk/aws-apigateway/lib/resource.ts

```diff
@@ -413,6 +424,14 @@ export class Resource extends ResourceBase {
       this.addCorsPreflight(this.defaultCorsPreflightOptions);
     }
   }
+
+  /**
+   * The RestApi associated with this Resource
+   * @deprecated - Throws an error if this Resource is not associated with an instance of `RestApi`. Use `api` instead.
+   */
+  public get restApi(): RestApi {
+    return this.parentResource!.restApi;
```

Comment:
Is `parentResource` promised to be defined?

Reply:
Yes, until it reaches `RootResource` where the implementation for `restApi` getter is implemented differently.

Reply:
I would still rather avoid the `!` as a rule of thumb and add explicitly throw an “unexpected” exception if null